types.go

```diff
@@ -277,10 +262,233 @@ type CheckResultsFilter struct {
+}
+
+//AlertChannelWebhook defines a type for a webhook alert channel
+type AlertChannelWebhook struct {
+	Name            string     `json:"name"`
+	Method          string     `json:"method"`
+	Headers         []KeyValue `json:"headers"`
+	QueryParameters []KeyValue `json:"queryParameters"`
+	Template        string     `json:"template"`
+	URL             string     `json:"url"`
+	WebhookSecret   string     `json:"webhookSecret"`
+}
+
+// AlertChannel represents an alert channel and its subscribed checks. The API
+// defines this data as read-only.
+type AlertChannel struct {
+	ID   int64  `json:"id,omitempty"`
+	Type string `json:"type"`
+	//Config        map[string]interface{} `json:"config"`
+	CreatedAt time.Time             `json:"created_at"`
+	UpdatedAt time.Time             `json:"updated_at"`
+	Email     *AlertChannelEmail    `json:"-"`
+	Slack     *AlertChannelSlack    `json:"-"`
+	SMS       *AlertChannelSMS      `json:"-"`
+	Opsgenie  *AlertChannelOpsgenie `json:"-"`
+	Webhook   *AlertChannelWebhook  `json:"-"`
+}
+
+//SetConfig sets config of alert channel based on it's type
+func (a *AlertChannel) SetConfig(cfg interface{}) {
+	switch v := cfg.(type) {
+	case *AlertChannelEmail:
+		a.Email = cfg.(*AlertChannelEmail)
+	case *AlertChannelSMS:
+		a.SMS = cfg.(*AlertChannelSMS)
+	case *AlertChannelSlack:
+		a.Slack = cfg.(*AlertChannelSlack)
+	case *AlertChannelWebhook:
+		a.Webhook = cfg.(*AlertChannelWebhook)
+	case *AlertChannelOpsgenie:
+		a.Opsgenie = cfg.(*AlertChannelOpsgenie)
+	default:
+		log.Printf("Unknown config type %v", v)
+	}
+}
+
+//GetConfig gets the config of the alert channel based on it's type
+func (a *AlertChannel) GetConfig() (cfg map[string]interface{}) {
+	byts := []byte{}
+	var err error
+	switch a.Type {
+	case AlertTypeEmail:
+		byts, err = json.Marshal(a.Email)
+	case AlertTypeSMS:
+		byts, err = json.Marshal(a.SMS)
+	case AlertTypeSlack:
+		byts, err = json.Marshal(a.Slack)
+	case AlertTypeWebhook:
+		byts, err = json.Marshal(a.Webhook)
+	case AlertTypeOpsgenie:
+		byts, err = json.Marshal(a.Opsgenie)
+	}
+
+	if err != nil {
+		panic(err)
+	}
+	err = json.Unmarshal(byts, &cfg)
+	return cfg
+}
+
+//AlertChannelConfigFromJSON gets AlertChannel.config from JSON
+func AlertChannelConfigFromJSON(channelType string, cfgJSON []byte) (interface{}, error) {
+	switch channelType {
+	case AlertTypeEmail:
+		r := AlertChannelEmail{}
+		json.Unmarshal(cfgJSON, &r)
+		return &r, nil
+	case AlertTypeSMS:
+		r := AlertChannelSMS{}
+		json.Unmarshal(cfgJSON, &r)
+		return &r, nil
+	case AlertTypeSlack:
+		r := AlertChannelSlack{}
+		json.Unmarshal(cfgJSON, &r)
+		return &r, nil
+	case AlertTypeOpsgenie:
+		r := AlertChannelOpsgenie{}
+		json.Unmarshal(cfgJSON, &r)
+		return &r, nil
+	case AlertTypeWebhook:
+		r := AlertChannelWebhook{}
+		json.Unmarshal(cfgJSON, &r)
+		return &r, nil
+	}
+	return nil, fmt.Errorf("Unknown AlertChannel.config type")
+}
+
+/*
+//GetEmailConfig gets config for an email alert channel
```

Comment:
Are these to be used later on?

Reply:
no, this was from the first version ! was refactored, should be removed